torch/csrc/jit/codegen/cuda/lower_shift.cpp

```diff
@@ -130,53 +130,31 @@ bool HaloInfo::hasRootAxisInfo(IterDomain* id) const {
   return root_axis_map_.find(id) != root_axis_map_.end();
 }
 
-bool HaloInfo::kirHasRootAxisInfo(IterDomain* id) const {
-  TORCH_INTERNAL_ASSERT(id->isKirStmt());
-  return kir_root_axis_map_.find(id) != kir_root_axis_map_.end();
-}
-
 const AxisHaloInfo& HaloInfo::getRootAxisInfo(IterDomain* id) const {
+  // TODO: Enable this check, was failing in many tests
```

Comment:
That isn't the case when `id` is from a reference tensor. In indexing, `updateHaloInfoForReference` propagates halo information of root domains of a consumer tensor to its reference tensor domains. When a consumer tensor is rfactored,  its root domain is effectively a rfactor domain of the reference tensor, but references don't have rfactor domains.

This could be changed to make it more robust. For example, `setRootAxisInfo` could have an optional parameter specifying whether a given `IterDomain` is from a reference tensor or not, and we could have the assertion check only for non-reference domains. Or, we might be able to have reference tensors to have rfactor information as well.